src/Education/XamlGenerator/XamlGeneratorHelper.cs

```diff
@@ -100,17 +111,97 @@ private void WriteExtendedDescriptionIfPresent(IRuleInfo ruleInfo)
+                    WriteSubTitleElement_Severity(ruleInfo);
+                }
 
-                WriteSubTitleElement_IssueType(ruleInfo);
-                WriteSubTitleElement_Severity(ruleInfo);
                 WriteSubTitleElement_RuleKey(ruleInfo);
                 WriteSubTitleElement_Tags(ruleInfo);
 
                 writer.WriteEndElement();
             }
 
+            #region Subtitle
+
+            #region New Clean Code Taxonomy
+
+            private void WriteCleanCodeHeader(IRuleInfo ruleInfo)
+            {
+                writer.WriteStartElement("BlockUIContainer");
+                writer.WriteStartElement("WrapPanel");
+                WriteCleanCodeHeader_CleanCodeAttribute(ruleInfo);
+                WriteCleanCodeHeader_SoftwareQualities(ruleInfo);
+                writer.WriteEndElement();
+                writer.WriteEndElement();
+            }
+
+            private void WriteCleanCodeHeader_CleanCodeAttribute(IRuleInfo ruleInfo)
+            {
+                var cleanCodeCategory = CleanCodeAttributeToCategoryMapping.Map[ruleInfo.CleanCodeAttribute.Value];
+                var cleanCodeAttribute = ruleInfo.CleanCodeAttribute;
+
+                WriteBubble(StyleResourceNames.CleanCodeAttributeBubble,
+                    () =>
+                    {
+                        WriteStylizedSpan(StyleResourceNames.CleanCodeSpan,
+                            () =>
+                            {
+                                WriteCleanCodeCategoryBold(cleanCodeCategory);
+                                writer.WriteString($" | Not {cleanCodeAttribute}");
+                            });
+                    });
+
+            }
+
+            private void WriteCleanCodeCategoryBold(CleanCodeCategory cleanCodeCategory)
+            {
+                writer.WriteStartElement("Run");
+                writer.ApplyStyleToElement(StyleResourceNames.CleanCodeCategory);
+                writer.WriteString(cleanCodeCategory.ToString());
+                writer.WriteEndElement();
+            }
+
+            private void WriteCleanCodeHeader_SoftwareQualities(IRuleInfo ruleInfo)
+            {
+                foreach (var softwareQualityAndSeverity in ruleInfo.DefaultImpacts)
+                {
+                    var imageInfo = SubTitleImageInfo.SoftwareQualitySeveritiesImages[softwareQualityAndSeverity.Value];
+
+                    WriteBubble(SoftwareQualityBubblesStyles[softwareQualityAndSeverity.Value],
+                        () =>
+                        {
+                            WriteSubTitleElement(softwareQualityAndSeverity.Key.ToString(), true);
+                            WriteSubTitleElementWithImage(imageInfo, true);
+                        });
+                }
+            }
+
+            private void WriteBubble(StyleResourceNames borderStyle, Action writeContent)
+            {
+                writer.WriteStartElement("Border");
```

Comment:
Probably worth a comment that the content of the `Border` will be kept together i.e. won't wrap onto separate lines.

Reply:
added the comment on the textbox